src/object/srv_obj.c

```diff
@@ -4006,8 +4009,8 @@ obj_obj_dtx_leader(struct dtx_leader_handle *dlh, void *arg, int idx,
 static int
 ds_obj_dtx_leader_prep_handle(struct daos_cpd_sub_head *dcsh,
 			      struct daos_cpd_sub_req *dcsrs,
-			      struct daos_shard_tgt *tgts, uint32_t tgt_cnt,
-			      uint32_t req_cnt, uint32_t *flags)
+			      struct daos_shard_tgt *tgts,
+			      int tgt_cnt, int req_cnt, uint32_t *flags)
```

Comment:
This solves this issue:

CID 306807 (#2 of 2): Negative loop bound (NEGATIVE_RETURNS) negative_returns: req_cnt is passed to a parameter that cannot be negative.
 
The negative return from ds_obj_cpd_get_dcsr_cnt() is :
        if (oci->oci_sub_reqs.ca_count <= dtx_idx)
                return -DER_INVAL;

should this error return be handled explicitly instead of passing to the for loop in ds_obj_dtx_leader_prep_handle()?
        for (i = 0; i < req_cnt; i++) {